internal/resources/cluster/integration/resource_integration_spec.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+var (
+	integrationSchema = map[string]*schema.Schema{
+		managementClusterNameKey: {
+			Type:        schema.TypeString,
+			Description: "Name of the management cluster",
+			Default:     attachedValue,
+			Optional:    true,
+			ForceNew:    true,
+		},
+
+		provisionerNameKey: {
+			Type:        schema.TypeString,
+			Description: "Provisioner of the cluster",
+			Default:     attachedValue,
+			Optional:    true,
+			ForceNew:    true,
+		},
+
+		clusterNameKey: {
+			Type:        schema.TypeString,
+			Description: "Name of this cluster",
+			Required:    true,
+			ForceNew:    true,
+		},
+
+		common.MetaKey: common.Meta,
+
+		specKey: {
+			Type:        schema.TypeList,
+			Description: "Specification for the Integration",
+			Required:    true,
+			ForceNew:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Elem:        specResource,
+		},
+
+		statusKey: {
+			Type:        schema.TypeMap,
+			Description: "Status of Integration",
+			Computed:    true,
+			Elem:        computedString,
+		},
+	}
+
+	specResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			integrationNameKey: {
+				Type:             schema.TypeString,
+				Description:      "Name of the Integration; valid options are currently only ['tanzu-service-mesh']",
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateIntegrationName,
+			},
+			configurationKey: {
+				Type:             schema.TypeString,
+				Description:      "Integration specific configurations in JSON format",
+				Optional:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateConfiguration,
+			},
+		},
```

Comment:
question, for enabling tanzu-service-mesh, we do not need secret or credentials reference ?
Just the configuration is sufficient ?

Reply:
The kubeconfig file for the cluster is sufficient